lib/puppet/type/file.rb

```diff
@@ -656,25 +677,18 @@ def recurse_remote(children)
       found = []
       total.reject! do |data|
         result = found.include?(data.relative_path)
-        found << data.relative_path unless found.include?(data.relative_path)
+        found << data.relative_path unless result
```

Comment:
@joshcooper When I split this into two commits, can this change happen in the first commit? Or does it have to wait until the second one. Not sure if it's a behavior change or not